### PR TITLE
add alpine x86 version

### LINKED ISSUE #5: Alpine version for x86 and ARM
Alpine version for x86 and ARM

Proposed solution by @Dashlorde 
- [ ] x86 #14 
- [ ] ARM #15

### PR DESCRIPTION
**Pull request by Dashlorde**
Fix #5

### CODE REVIEW

File: 3.4/x86_alpine/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+COPY ./start.sh /opt/apache2/start.sh
+
+RUN apk update && apk upgrade && \
+    apk add --no-cache wget unzip apache2 php7 postgresql-dev \
+    php7-opcache php7-pgsql php7-xmlrpc php7-gd php7-intl php7-session php7-json \
+    php7-iconv php7-xml php7-apache2 php7-curl php7-soap php7-openssl php7-ctype php7-zip \
+    php7-zlib php7-simplexml php7-dom php7-xmlreader php7-fileinfo php7-mbstring php7-tokenizer
+
+RUN wget -O /var/www/moodle-${version}.tar.gz https://github.com/moodle/moodle/archive/v${version}.tar.gz && \
```
Comment (empeje) [inline 3.4/x86_alpine/Dockerfile]: could you delete the artifact after extraction?

File: 3.4/x86_alpine/config/php.ini [hunk 2]
```diff
@@ -0,0 +1,1936 @@
+[PHP]
```
Comment (empeje) [inline 3.4/x86_alpine/config/php.ini]: Can you minify this by delete comment?

File: 3.4/x86_alpine/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+moodledb_alpine:
+  image: postgres:10
+  container_name: moodledb_alpine
+  ports:
+    - "5432:5432"
+  environment:
+    - POSTGRES_DATABASE=moodle
```
Comment (empeje) [inline 3.4/x86_alpine/docker-compose.yml]: Put comment here like #16

File: 3.4/x86_alpine/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    - POSTGRES_PASSWORD=moodle
+moodle_alpine:
+  build: .
+  container_name: moodle_alpine
+  ports:
+    - "80:80"
+  links:
+    - moodledb_alpine
+  environment:
+    - MOODLE_URL=http://127.0.0.1:80
```
Comment (empeje) [inline 3.4/x86_alpine/docker-compose.yml]: We have new convention see #16 and add also config like this https://github.com/ole-vi/moodole/blob/e65dd944d14e26a802364bfab8131a77af8e2623/3.4/x86/config.php

**Review by empeje — CHANGES_REQUESTED** [review]
Need some changes, due to use case requirement

* Need to access site from different way
  - IP
  - Domain
  - local
  - Tunnel

follow convention written here #16


@Dashlorde how many MB the resulted image would be?

**Comment by empeje** [comment]
Please adapt with this changes https://github.com/ole-vi/moodole/pull/26

**Comment by empeje** [comment]
Oops, you forgot about adding travis build for x86_alpine. I'll add one for you.

File: 3.4/x86_alpine/start.sh [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+if [ ! -z "${MOODOLE_DB_PORT}" ]
+then
+  echo "env[MOODOLE_DB_PORT] = '$MOODOLE_DB_PORT'" >> $PHP_FM_CONFIG
+fi
+
+/usr/sbin/php-fpm7
+
+/usr/sbin/nginx
+
+tail -f /var/log/nginx/access.log
```
Comment (empeje) [inline 3.4/x86_alpine/start.sh:49]: I think this log address is not correct, I didn't see any logs redirected here 😉

**Review by empeje — CHANGES_REQUESTED** [review]
screen shot 2018-01-18 at 11 10 34 pm
screen shot 2018-01-18 at 11 07 09 pm

It is working, only logs missing.

**Comment by empeje** [comment]
@dogi Please take a look if you have time

File: 3.4/x86_alpine/start.sh (hunk 5 above)
Comment (Dashlorde) [reply 3.4/x86_alpine/start.sh:49]: But I see logs in /var/log/nginx/access.log in moodle container. I am confused
screen shot 2018-01-18 at 12 25 14 pm

Comment (empeje) [reply 3.4/x86_alpine/start.sh:49]: have you tried `docker log -f 7814143b6f05`?

**Comment by empeje** [comment]
Good job @Dashlorde , now I can see the log redirected to stdout 
screen shot 2018-01-20 at 3 03 23 am

Do you have any addition to this PR @Dashlorde ? If not, I will merge this one.

**Review by empeje — APPROVED** [review]

**Comment by Dashlorde** [comment]
no additional pr so far @mappuji

**Comment by empeje** [comment]
add @sente here

**Comment by empeje** [comment]
@Dashlorde please consider sente https://github.com/treehouses/moodole/pull/32 comments

https://docs.splunk.com/Documentation/AddOns/released/NGINX/Setupv2

**Comment by Dashlorde** [comment]
@mappuji I edit the `/etc/nginx/nginx.conf` to stdout the log. The access log is already set to `/var/log/nginx/access.log` by default.

**Comment by empeje** [comment]
Awesome, I'll verify and merge it.
Greate @Dashlorde !

Can we continue to #15 ?

**Comment by Dashlorde** [comment]
updated #15 @mappuji

**Comment by empeje** [comment]
Thanks, awesome @Dashlorde

**Review by empeje — CHANGES_REQUESTED** [review]
Please fix it

**Comment by Dashlorde** [comment]
fixed, please review @mappuji

**Review by empeje — APPROVED** [review]

**Comment by empeje** [comment]
I approved your changes, will merge it.

**Comment by empeje** [comment]
@dogi finally we ship it, the arm alpine is ongoing, just need quick fix on the travis pipeline.

**Pull request merged**